aws/src/main/java/org/apache/iceberg/aws/AwsClientProperties.java

```diff
@@ -154,6 +156,23 @@ public AwsCredentialsProvider credentialsProvider(
     return DefaultCredentialsProvider.builder().build();
   }
 
+  /**
+   * Set ADAPTIVE_V2 <a
+   * href="https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/core/retry/RetryMode.html">RetryMode</a>
+   * for an AWS client builder.
+   *
+   * <p>For more details, see
+   * https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/core/client/config/ClientOverrideConfiguration.Builder.html#retryStrategy(software.amazon.awssdk.core.retry.RetryMode)
+   */
```

Comment:
Could we structure this JavaDoc similar to the  the other apply configuration methods? e.g. checkout the java doc in applyClientCredentialConfigurations or applyClientRegionConfiguration